### ISSUE TITLE
Azure Service Fabric Cluster won't start.

### ISSUE THREAD

**Issue by goochjs** [Azure/service-fabric-issues#862]
Hi,

I'm creating a set of AzureRM scripts to deploy a Service Fabric cluster.  The VM scale set creates successfully but the Service Fabric software fails to install.

From the Windows Event Viewer, I can see a number of warnings and errors.  Here's an example:-

> Log Name:      Application
> Source:        ServiceFabricNodeBootstrapAgent
> Date:          2/19/2018 11:02:09 AM
> Event ID:      0
> Task Category: None
> Level:         Warning
> Keywords:      Classic
> User:          N/A
> Computer:      cluster000000
> Description:
> Installer is either a not supported installer format or the product did not exist on this machine.
> Event Xml:
> `<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">`
>   `<System>`
>     `<Provider Name="ServiceFabricNodeBootstrapAgent" />`
>     `<EventID Qualifiers="0">0</EventID>`
>     `<Level>3</Level>`
>     `<Task>0</Task>`
>     `<Keywords>0x80000000000000</Keywords>`
>     `<TimeCreated SystemTime="2018-02-19T11:02:09.709805300Z" />`
>     `<EventRecordID>568</EventRecordID>`
>     `<Channel>Application</Channel>`
>     `<Computer>cluster000000</Computer>`
>     `<Security />`
>   `</System>`
>   `<EventData>`
>     `<Data>Installer is either a not supported installer format or the product did not exist on this machine.</Data>`
>   `</EventData>`
> `</Event>`

It then seems to loop around trying to install the software before getting stuck into a repeating set of errors:-

> Log Name:      Application
> Source:        ServiceFabricNodeBootstrapAgent
> Date:          2/19/2018 11:07:56 AM
> Event ID:      0
> Task Category: None
> Level:         Error
> Keywords:      Classic
> User:          N/A
> Computer:      cluster000000
> Description:
> Failed starting service, Error: Microsoft.Azure.ServiceFabric.Extension.Core.AgentException: Configure node failed with code -1
>    at Microsoft.Azure.ServiceFabric.Extension.Core.NodeBootstrapAgent.StartFabricNode(Boolean isBootstrapping)
> Event Xml:
> `<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">`
>   `<System>`
>     `<Provider Name="ServiceFabricNodeBootstrapAgent" />`
>     `<EventID Qualifiers="0">0</EventID>`
>     `<Level>2</Level>`
>     `<Task>0</Task>`
>     `<Keywords>0x80000000000000</Keywords>`
>     `<TimeCreated SystemTime="2018-02-19T11:07:56.104664600Z" />`
>     `<EventRecordID>905</EventRecordID>`
>     `<Channel>Application</Channel>`
>     `<Computer>cluster000000</Computer>`
>     `<Security />`
>   `</System>`
>   `<EventData>`
>     `<Data>Failed starting service, Error: Microsoft.Azure.ServiceFabric.Extension.Core.AgentException: Configure node failed with code -1
>    `at Microsoft.Azure.ServiceFabric.Extension.Core.NodeBootstrapAgent.StartFabricNode(Boolean isBootstrapping)</Data>`
>   `</EventData>`
> `</Event>`

I can also see a number of WER files, one for every time Windows tries to start the service.  They contain this:-

> Version=1
> EventType=APPCRASH
> EventTime=131635245218671902
> ReportType=2
> Consent=1
> UploadTime=131635245220244746
> ReportIdentifier=1cd1cdcd-1582-11e8-a94c-000d3a215ee8
> IntegratorReportIdentifier=9b5b6b29-03a1-4fa7-aa48-8cd3c8406772
> NsAppName=FabricInstallerService.exe
> AppSessionGuid=00000cdc-0000-0014-9aa1-03df8ea9d301
> TargetAppId=W:0006f6ea17540e1d126351789d0bbd77d53900000904!0000b8fc4083764c64b947dd3c9d0c6470d75da97cb9!FabricInstallerService.exe
> TargetAppVer=2018//01//22:23:13:00!691982!FabricInstallerService.exe
> BootId=4294967295
> Response.BucketId=e26fa10405c9a539e645fee0cce15245
> Response.BucketTable=4
> Response.LegacyBucketId=1604969083690897989
> Response.type=4
> Sig[0].Name=Application Name
> Sig[0].Value=FabricInstallerService.exe
> Sig[1].Name=Application Version
> Sig[1].Value=6.1.456.9494
> Sig[2].Name=Application Timestamp
> Sig[2].Value=5a666ffc
> Sig[3].Name=Fault Module Name
> Sig[3].Value=FabricInstallerService.exe
> Sig[4].Name=Fault Module Version
> Sig[4].Value=6.1.456.9494
> Sig[5].Name=Fault Module Timestamp
> Sig[5].Value=5a666ffc
> Sig[6].Name=Exception Code
> Sig[6].Value=c0000602
> Sig[7].Name=Exception Offset
> Sig[7].Value=000000000016fc4d
> DynamicSig[1].Name=OS Version
> DynamicSig[1].Value=10.0.14393.2.0.0.400.8
> DynamicSig[2].Name=Locale ID
> DynamicSig[2].Value=1033
> DynamicSig[22].Name=Additional Information 1
> DynamicSig[22].Value=efe1
> DynamicSig[23].Name=Additional Information 2
> DynamicSig[23].Value=efe1d3ca6f5d5bb6e915fdc66edacaa4
> DynamicSig[24].Name=Additional Information 3
> DynamicSig[24].Value=d938
> DynamicSig[25].Name=Additional Information 4
> DynamicSig[25].Value=d938b4f18e2dc45e2f0a49496c2aaaa0
> UI[2]=C:\Program Files\Microsoft Service Fabric\FabricInstallerService.Code\FabricInstallerService.exe
> UI[5]=Check online for a solution (recommended)
> UI[6]=Check for a solution later (recommended)
> UI[7]=Close
> UI[8]=FabricInstallerService stopped working and was closed
> UI[9]=A problem caused the application to stop working correctly. Windows will notify you if a solution is available.
> UI[10]=&Close
> LoadedModule[0]=C:\Program Files\Microsoft Service Fabric\FabricInstallerService.Code\FabricInstallerService.exe
> LoadedModule[1]=C:\Windows\SYSTEM32\ntdll.dll
> LoadedModule[2]=C:\Windows\System32\KERNEL32.DLL
> LoadedModule[3]=C:\Windows\System32\KERNELBASE.dll
> LoadedModule[4]=C:\Windows\System32\ADVAPI32.dll
> LoadedModule[5]=C:\Windows\System32\msvcrt.dll
> LoadedModule[6]=C:\Windows\System32\sechost.dll
> LoadedModule[7]=C:\Windows\System32\RPCRT4.dll
> LoadedModule[8]=C:\Windows\System32\USER32.dll
> LoadedModule[9]=C:\Windows\System32\win32u.dll
> LoadedModule[10]=C:\Windows\System32\GDI32.dll
> LoadedModule[11]=C:\Windows\System32\gdi32full.dll
> LoadedModule[12]=C:\Windows\System32\ucrtbase.dll
> LoadedModule[13]=C:\Windows\System32\CRYPT32.dll
> LoadedModule[14]=C:\Windows\System32\MSASN1.dll
> LoadedModule[15]=C:\Windows\System32\WS2_32.dll
> LoadedModule[16]=C:\Windows\System32\SHLWAPI.dll
> LoadedModule[17]=C:\Windows\SYSTEM32\RstrtMgr.DLL
> LoadedModule[18]=C:\Windows\System32\combase.dll
> LoadedModule[19]=C:\Windows\System32\ole32.dll
> LoadedModule[20]=C:\Windows\System32\bcryptPrimitives.dll
> LoadedModule[21]=C:\Windows\System32\OLEAUT32.dll
> LoadedModule[22]=C:\Windows\System32\msvcp_win.dll
> LoadedModule[23]=C:\Program Files\Microsoft Service Fabric\FabricInstallerService.Code\FabricResources.dll
> LoadedModule[24]=C:\Program Files\Microsoft Service Fabric\FabricInstallerService.Code\FabricCommon.dll
> LoadedModule[25]=C:\Program Files\Microsoft Service Fabric\FabricInstallerService.Code\MSVCP140.dll
> LoadedModule[26]=C:\Program Files\Microsoft Service Fabric\FabricInstallerService.Code\VCRUNTIME140.dll
> LoadedModule[27]=C:\Windows\SYSTEM32\IPHLPAPI.DLL
> LoadedModule[28]=C:\Windows\SYSTEM32\Secur32.dll
> LoadedModule[29]=C:\Windows\SYSTEM32\dbghelp.dll
> LoadedModule[30]=C:\Windows\SYSTEM32\USERENV.dll
> LoadedModule[31]=C:\Windows\System32\profapi.dll
> LoadedModule[32]=C:\Windows\SYSTEM32\ncrypt.dll
> LoadedModule[33]=C:\Program Files\Microsoft Service Fabric\FabricInstallerService.Code\FabricAad.Client.dll
> LoadedModule[34]=C:\Windows\SYSTEM32\NETAPI32.dll
> LoadedModule[35]=C:\Windows\SYSTEM32\VERSION.dll
> LoadedModule[36]=C:\Windows\SYSTEM32\Cabinet.dll
> LoadedModule[37]=C:\Windows\SYSTEM32\XmlLite.dll
> LoadedModule[38]=C:\Windows\SYSTEM32\bcrypt.dll
> LoadedModule[39]=C:\Windows\SYSTEM32\NTASN1.dll
> LoadedModule[40]=C:\Windows\SYSTEM32\mscoree.dll
> LoadedModule[41]=C:\Windows\SYSTEM32\CRYPTBASE.DLL
> LoadedModule[42]=C:\Windows\System32\WINTRUST.dll
> LoadedModule[43]=C:\Windows\SYSTEM32\SSPICLI.DLL
> LoadedModule[44]=C:\Windows\SYSTEM32\DSROLE.DLL
> LoadedModule[45]=C:\Windows\Microsoft.NET\Framework64\v4.0.30319\mscoreei.dll
> LoadedModule[46]=C:\Windows\System32\kernel.appcore.dll
> State[0].Key=Transport.DoneStage1
> State[0].Value=1
> FriendlyEventName=Stopped working
> ConsentKey=APPCRASH
> AppName=FabricInstallerService
> AppPath=C:\Program Files\Microsoft Service Fabric\FabricInstallerService.Code\FabricInstallerService.exe
> NsPartner=windows
> NsGroup=windows8
> ApplicationIdentity=5972214350E10A57CC79C13EB2A13A82
> MetadataHash=462872607

I'm not sure where I should look for more diagnostic information.

My JSON file can be seen here -> https://github.com/goochjs/azure-testbed/blob/master/service-fabric/az-servicefabric.json

Thanks for any advice.

J.

**Comment by mhatreabhay** [comment]
Can you share following details. 
1. subscription id
2. resource id
3. region/location

**Comment by goochjs** [comment]
Hi - thanks for the reply.  In last couple of days, we've tracked the problem down to the Service Fabric cluster being configured to use storage accounts.  We've changed the config to use managed disks and it came up successfully.

We eventually managed to work this out after finding the trace files held under **D:\SvcFab\Log\Traces** (example copied below for others people's reference), which gave us a clue that the machines couldn't find their storage.

J.

> 2018/02/27-10:35:26.817,Info,3568,FabricDeployer.FabricDeployer,Running deployer with Configure /fabricBinRoot: /fabricDataRoot:D:\SvcFab /fabricLogRoot:D:\SvcFab\Log /cm:C:\WindowsAzure\Logs\Plugins\Microsoft.Azure.ServiceFabric.ServiceFabricNode\1.0.0.37\TempClusterManifest.xml /oldClusterManifestString: /im:C:\WindowsAzure\Logs\Plugins\Microsoft.Azure.ServiceFabric.ServiceFabricNode\1.0.0.37\InfrastructureManifest.xml /instanceId: /targetVersion: /nodeName: /nodeTypeName: /runAsType: /runAsAccountName: /runAsPassword: /serviceStartupType: /output: /currentVersion: /error: /bootstrapMSIPath: /machineName: /fabricPackageRoot: /jsonClusterConfigLocation: /enableCircularTraceSession:False /continueIfContainersFeatureNotInstalled: /skipDeleteData:
2018/02/27-10:35:26.825,Info,3568,ImageStoreClient.ManagedFileLock,Obtained writer lock for D:\SvcFab\lock
2018/02/27-10:35:26.832,Info,3568,FabricDeployer.FabricDeployer,Executing Configure /fabricBinRoot: /fabricDataRoot:D:\SvcFab /fabricLogRoot:D:\SvcFab\Log /cm:C:\WindowsAzure\Logs\Plugins\Microsoft.Azure.ServiceFabric.ServiceFabricNode\1.0.0.37\TempClusterManifest.xml /oldClusterManifestString: /im:C:\WindowsAzure\Logs\Plugins\Microsoft.Azure.ServiceFabric.ServiceFabricNode\1.0.0.37\InfrastructureManifest.xml /instanceId: /targetVersion: /nodeName: /nodeTypeName: /runAsType: /runAsAccountName: /runAsPassword: /serviceStartupType: /output: /currentVersion: /error: /bootstrapMSIPath: /machineName: /fabricPackageRoot: /jsonClusterConfigLocation: /enableCircularTraceSession:False /continueIfContainersFeatureNotInstalled: /skipDeleteData:
2018/02/27-10:35:26.911,Info,3568,FabricDeployer.FabricDeployer,Running operation System.Fabric.FabricDeployer.ConfigureOperation
2018/02/27-10:35:26.931,Info,3568,FabricDeployer.FabricDeployer,Creating FabricDataRoot D:\SvcFab, if it doesn't exist on machine
2018/02/27-10:35:26.931,Info,3568,FabricDeployer.FabricDeployer,Creating FabricLogRoot D:\SvcFab\Log, if it doesn't exist on machine
2018/02/27-10:35:26.935,Info,3568,FabricDeployer.FabricDeployer,The current machine IP addresses are: fe80::bc75:1059:4ab8:bdb%4, fe80::c59a:9841:d9a7:1fc2%6, 10.0.0.4, 172.27.112.1, 127.0.0.1, ::1
2018/02/27-10:35:27.034,Error,3568,ImageBuilder.AzureBlobServiceFabricEtw,Section AzureBlobServiceFabricEtw, parameter StoreConnectionString: Failed to retrieve protected account key named StorageAccountKey1..
2018/02/27-10:35:27.040,Error,3568,ImageBuilder.AzureTableServiceFabricEtwQueryable,Section AzureTableServiceFabricEtwQueryable, parameter StoreConnectionString: Failed to retrieve protected account key named StorageAccountKey1..
2018/02/27-10:35:27.044,Error,3568,ImageBuilder.AzureBlobServiceFabricCrashDump,Section AzureBlobServiceFabricCrashDump, parameter StoreConnectionString: Failed to retrieve protected account key named StorageAccountKey1..
2018/02/27-10:35:27.045,Error,3568,ImageBuilder.AzureBlobServiceFabricPerfCounter,Section AzureBlobServiceFabricPerfCounter, parameter StoreConnectionString: Failed to retrieve protected account key named StorageAccountKey1..
2018/02/27-10:35:27.050,Info,3568,ImageStoreClient.ManagedFileLock,Released writer lock on D:\SvcFab\lock
2018/02/27-10:35:27.053,Error,3568,FabricDeployer.FabricDeployer,System.Fabric.FabricDeployer.ClusterManifestValidationException: Cluster manifest validation failed with exception System.ArgumentException: DCA settings validator failed
	   at System.Fabric.Management.WindowsFabricValidator.WindowsFabricSettings..ctor(ClusterManifestType manifest, Boolean skipValidation)
	   at System.Fabric.FabricDeployer.FabricValidatorWrapper.ValidateAndEnsureDefaultImageStore()
	   at System.Fabric.FabricDeployer.FabricValidatorWrapper.ValidateAndEnsureDefaultImageStore()
	   at System.Fabric.FabricDeployer.ConfigureOperation.OnExecuteOperation(DeploymentParameters parameters, ClusterManifestType clusterManifest, Infrastructure infrastructure)
	   at System.Fabric.FabricDeployer.DeploymentOperation.ExecuteOperationPrivate(DeploymentParameters parameters)
	   at System.Fabric.FabricDeployer.DeploymentOperation.ExecuteOperation(DeploymentParameters parameters, Boolean disableFileTrace)
	   at System.Fabric.FabricDeployer.Program.Main(String[] args)
